### PR TITLE
Fix some stuff

### LINKED ISSUE #114: Tapping a notification while the app is closed does not open the activity
The first notification you tap opens the app but doesn't actually bring up a view activity screen. If you immediately tap on another notification, it'll take you to the screen. This is probably because you have to handle those two cases differently in code, and something must have changed for handling the case for when the app is fully closed

### LINKED ISSUE #127: App crashes when notification's activity is deleted
1. Create an activity on a different device
2. Wait until your real device gets a notification 
3. Delete the activity from your different device.
4. Press the notification
5. Crash!

### LINKED ISSUE #131: Name field in set up account screen
If you enter a name in sign up, it goes away once you get to the set up account view.  
Screen Shot 2019-04-04 at 11 00 02 AM
Screen Shot 2019-04-04 at 11 00 08 AM

Then, when you fill out a bio and profile picture, it says you're missing a field, since name is not entered.

cc @JakeThurman who added the duplication of the field

### PR DESCRIPTION
**Pull request by JakeThurman**
I don't have a way to test notifications, but other fixes should be good.

Fixes #131 
Fixes #127 (probably) (EDIT:confirmed)
Fixes #114

To Test (131)
- Create an Account, see confirm password work.

To Test (127)
- Setup two devices in the same location (and appropriate favorite topics).
- Create an activity.
- Delete this activity.
- Open notification received by the other device.
- Observe app (probably) not crashing.

### CODE REVIEW

File: Buddies/Activities/View Activity/ViewActivityController.swift [hunk 1]
```diff
@@ -202,9 +202,10 @@ class ViewActivityController: UIViewController {
     func loadWith(
         _ activityId: ActivityId?,
         dataAccess: DataAccessor = DataAccessor.instance,
-        currentUser uid: String = Auth.auth().currentUser!.uid
+        currentUser uid: String? = Auth.auth().currentUser?.uid
```
Comment (noahtallen) [inline Buddies/Activities/View Activity/ViewActivityController.swift:204]: lol good call

File: Buddies/Activities/View Activity/ViewActivityController.swift [hunk 2]
```diff
@@ -303,7 +304,7 @@ class ViewActivityController: UIViewController {
                 // before this is called. The else case should never be called.
                 // Insert it underneath the description view in the hierarchy.
                 if let desc = self.descriptionView {
-                    contentArea.insertSubview((chatController.view)!, belowSubview: desc)
```
Comment (noahtallen) [inline Buddies/Activities/View Activity/ViewActivityController.swift:334]: 🤔

**Review by noahtallen — COMMENTED** [review]
What specifically should solve the notification issue?

**Comment by JakeThurman** [comment]
@noahtallen tbh... no idea. 
I didn't have a device to test so I looked for potential issues I could see and corrected them. Hoping for the best. If you can test and see the issue still, if you just give me the line that failed I can look into it more.

**Review by noahtallen — APPROVED** [review]
I tested, and I did not get a crash when opening the app from a notification for a deleted activity! also, the new login looks good

**Pull request merged**